### PR TITLE
Add xyzspaces

### PR DESCRIPTION
**Pull request by deeplook**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by deeplook** [comment]
Since I am posting this PR I am also willing to be listed here... ;)

**Comment by deeplook** [comment]
I see in the Azure pipeline logs that `fiona` is used, but not declared as a requirement. Which is weird because on Travis-CI there were building fine. Maybe the conda builds are in some sense stricter, and if so I'd like to know in which one?

File: recipes/xyzspaces/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/heremaps/xyz-spaces-python/archive/{{ version }}.tar.gz
+  sha256: 060f023a9711d0d6abe1381ec2723ee8ec0c6551f24a413098295bf8b58121ed
+
+build:
+  number: 0
+  skip: True  # [py<36]
```
Comment (synapticarbors) [inline recipes/xyzspaces/meta.yaml:14]: conda-forge isn't building for pytho<3.6, so you can remove the skip and use `noarch: python`:

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

you can always add the version restrictions under `requirements` (i.e change `python` to `python  >=3.6` under `host` and `run`)

File: recipes/xyzspaces/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
```
Comment (synapticarbors) [inline recipes/xyzspaces/meta.yaml:22]: You need to add everything in the `setup.py`'s `install_requires`, which looks to be read out of:

https://github.com/heremaps/xyz-spaces-python/blob/master/requirements.txt

I'm guessing the issue with `fiona` not being on the list, but being required is that it is a dependency of one of the listed dependencies, so it gets picked up. Just a guess though

**Comment by deeplook** [comment]
I don't seem to put the pip-only dependencies like "turfpy" and "geobuf" at the right place. When trying to run `conda smithy recipe-lint .` locally I get "No azure token. Create a token and put it in ~/.conda-smithy/azure.token. Running out of ideas...

**Comment by synapticarbors** [comment]
@deeplook - we don't really want any pip-only dependencies. We'll need to open separate pull requests for `turfpy` and `geobuf` and get those merged before we can move forward with this recipe.

**Comment by deeplook** [comment]
@synapticarbors While I can imaging the reasoning behind not wanting to support plain pip packages (which _is_ somewhat ironic, as pip is still needed, I believe), I have failed to see a bold argument like this made on https://conda-forge.org/docs/index.html. I think this could help people spend more time on relevant things (like convincing other maintainers to publish their packages on conda-forge). What do you think?

**Comment by synapticarbors** [comment]
@deeplook -- I've submitted a pull request (https://github.com/conda-forge/staged-recipes/pull/12514) to address the missing dependencies.

**Comment by deeplook** [comment]
I see https://github.com/conda-forge/turfpy-feedstock and https://github.com/conda-forge/geobuf-feedstock now. Awesome, thanks!

**Comment by synapticarbors** [comment]
@deeplook -- I restarted the build in this PR and it looks like linux is now passing. Just waiting on the others.

**Comment by synapticarbors** [comment]
@deeplook -- can you confirm that the other co-maintainers have agreed to be listed? Otherwise they should comment here saying they are.

**Comment by deeplook** [comment]
@synapticarbors I confirm.

**Comment by deeplook** [comment]
@synapticarbors Anything else I need to do? Otherwise this is ready for review.

**Comment by deeplook** [comment]
Thanks!

**Comment by deeplook** [comment]
@synapticarbors BTW, the description field on https://anaconda.org/conda-forge/xyzspaces is empty despite the description given in the `meta.yaml` file. Am I missing something?

**Comment by synapticarbors** [comment]
No packages on conda-forge have that populated (e.g https://anaconda.org/conda-forge/numpy). The description shows up in https://github.com/conda-forge/xyzspaces-feedstock#about-xyzspaces

**Pull request merged**